### PR TITLE
Cleanup scheduler/algorithm/predicates package

### LINKED ISSUE #86993: Remove scheduler/algorithm/priorities/util package [Migration Phase 2]
The logic should move to the plugins using the util logic, or to plugins/helper package if it is shared with multiple plugins.

/assign @Huang-Wei 
/sig scheduling
/priority important-soon
Part of #85822

### PR DESCRIPTION
**Pull request by Huang-Wei**
**What type of PR is this?**

/kind cleanup
/sig scheduling

**What this PR does / why we need it**:

The only leftover under predicates/util.go is `PortsConflict`, its logic is moved inline into predicates.go, and the whole PodFitsHostPortsPredicate will move to nodeports plugin later.

**Which issue(s) this PR fixes**:

Part of #86712

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

/cc @ahg-g

### CODE REVIEW

**Comment by ahg-g** [comment]
hmm,  the issue was about priorities/util, but we need this as well.

File: pkg/scheduler/framework/plugins/helper/node_ports.go [hunk 1]
```diff
@@ -14,16 +14,17 @@ See the License for the specific language governing permissions and
...
 
-package predicates
+package helper
 
 import (
 	v1 "k8s.io/api/core/v1"
 	schedulernodeinfo "k8s.io/kubernetes/pkg/scheduler/nodeinfo"
 )
 
-// portsConflict check whether existingPorts and wantPorts conflict with each other
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/helper/node_ports.go]: we don't want to introduce more cleanup backlog, lets move this logic to predicates.go, just place it  inside ```PodFitsHostPortsPredicate```, there is no need for a separate function, and then the whole ```PodFitsHostPortsPredicate``` will move to nodeports plugin later.

**Comment by ahg-g** [comment]
also, while we are cleaning the predicates pkg, can you move the ```Metadata``` type to predicates.go and delete the metadata.go file please.

**Comment by Huang-Wei** [comment]
> hmm, the issue was about priorities/util, but we need this as well.

oops... I misread. Updated the PR description and commit.

**Pull request merged**